docs/markdown/GuiTutorial.md

```diff
@@ -0,0 +1,194 @@
+---
+short-description: A simple GUI tutorial
+...
+
+# Building a simple SDL2 app from scratch
+
+This page shows from the ground up how to define and build an SDL2 gui
+application using nothing but Meson. The document is written for
+Windows, as it is the most complex platform, but the same basic ideas
+should work on Linux and macOS as well.
+
+The sample application is written in plain C as SDL 2 is also written
+in C. If you prefer C++ instead, the conversion is fairly simple and
+is left as an exercise to the reader.
+
+This document assumes that you have already installed both Visual
+Studio and Meson.
+
+# Set up the build directory
+
+First you need to create an empty directory for all your stuff. The
+Visual Studio toolchain is a bit unusual in that it requires you to
+run your builds from a specific shell. This can be found by opening
+the application menu and then choosing `Visual Studio <year> -> x86_64
+native tools command prompt`.
+
+It will put you in a weird directory, so you need to go to your home
+directory:
+
+    cd \users\yourusername
+
+Typically you'd type `cd \users\` and then press the tabulator key to
+make the shell autocomplete the username. Once that is done you can
+create the directory.
+
+    mkdir sdldemo
+    cd sdldemo
+
+# Creating the sample program
+
+Now we need to create a source file and a Meson build definition file.
+We're not going to use SDL at all, but instead start with a simple
+program that only prints some text. Once we have it working we can
+extend it to do graphics. The source goes into a file `sdlprog.c` and
+has the following contents:
+
+```c
+#include <stdio.h>
+
+int main(int argc, char **argv) {
+  printf("App is running.\n");
+  return 0;
+}
+```
+
+The build definition goes to a file called `meson.build` and looks like this:
+
+```meson
+project('sdldemo', 'c')
+
+executable('sdlprog', 'sdlprog.c')
+```
+
+With this done we can start the build with the following command:
+
+    meson build
```

Comment:
It's somewhat customary in our docs to refer to `builddir` to make it more obvious that it is not a subcommand, but a directory name.

Reply:
Updated.